### ISSUE TITLE
[Controller doesn't work after 1.5.0 Patch ] [MGS3]: 

### ISSUE THREAD

**Issue by FCGre** [Lyall/MGSHDFix#97]
i have xbox controller and its doesn't work after 1.5.0 update.

**Comment by mooms06** [comment]
Update 1.5.0 broke all controller support except Steam Input, for all games.

**Comment by FCGre** [comment]
what's the solution?

**Comment by mooms06** [comment]
> what's the solution?

Rollback to 1.4.1 or wait for a patch.

**Comment by ShizCalev** [comment]
Patch is out, verify your game files and it'll be fixed.

**Comment by LethalPlacebo** [comment]
> Patch is out, verify your game files and it'll be fixed.

Tangentially related, but I and many others have now lost the vibration function in MGS2 and MGS3. My controllers work just fine, but there's no longer any vibration. Could it be related to this? Also, what patch released, I'm not seeing a new one? I tried verifying the files, making sure Steam Input is enabled, and making sure I'm using the official controller template, nothing seemed to bring it back.

**Comment by ShizCalev** [comment]
> > Patch is out, verify your game files and it'll be fixed.
> 
> Tangentially related, but I and many others have now lost the vibration function in MGS2 and MGS3. My controllers work just fine, but there's no longer any vibration. Could it be related to this? Also, what patch released, I'm not seeing a new one? I tried verifying the files, making sure Steam Input is enabled, and making sure I'm using the official controller template, nothing seemed to bring it back.

https://steamdb.info/changelist/22740193/

![image](https://github.com/Lyall/MGSHDFix/assets/6209658/4126e4d1-58aa-46de-8f60-b998c4270f67)

Sorry, inconsistent info. It's unfortunately a bug with steam input where it's not updating existing configs correctly. 

Appears Konami's trying to unfuck it right now
![image](https://github.com/Lyall/MGSHDFix/assets/6209658/3cdcd8be-0f52-4511-b616-50e356050886)

In the mean time you can try to force your controller configs to update with these links

MGS2
XBONE
steam://controllerconfig/2131640/3175039636
PS4 
steam://controllerconfig/2131640/3175039358
PS5 
steam://controllerconfig/2131640/3175039052
switch 
steam://controllerconfig/2131640/3175039827

MGS3
XBONE
steam://controllerconfig/2131650/3165224769
PS4 
steam://controllerconfig/2131650/3165224429
PS5 
steam://controllerconfig/2131650/3165223759
switch 
steam://controllerconfig/2131650/3165223491

**Comment by LethalPlacebo** [comment]
> Sorry, inconsistent info. It's unfortunately a bug with steam input where it's not updating existing configs correctly.
> 
> Appears Konami's trying to unfuck it right now 
Tried the configs for both games, still no vibration though again my controller works just fine, thank you anyway. Good to know Konami is already aware of the issue and trying to fix it as well.

**Comment by LethalPlacebo** [comment]
Tried the configs for both games, still no vibration though again my controller works just fine, thank you anyway. Good to know Konami is already aware of the issue and trying to fix it as well.

**Comment by LethalPlacebo** [comment]
Vibration is fixed. Newest patch, about 10MB big for each game, fixed the vibration issue. Native controller support is still gone, meaning the game still requires Steam Input.

**Comment by lazymanzy** [comment]
Hi! Any chance this controller issue is fixable for version 1.5.0? I dont care much for the vibration thing just that the controller doesnt work at all on this verison. I only have the version 1.5.0 for *obvious* reasons and I cannot find the 1.5.1 version anywhere plus Im not even sure it would fix anything. I use DS4Win and PS4 controller for the Master Collection and it worked well with MGS1... Thanks for any help.

**Comment by LethalPlacebo** [comment]
> Hi! Any chance this controller issue is fixable for version 1.5.0? I dont care much for the vibration thing just that the controller doesnt work at all on this verison. I only have the version 1.5.0 for _obvious_ reasons and I cannot find the 1.5.1 version anywhere plus Im not even sure it would fix anything. I use DS4Win and PS4 controller for the Master Collection and it worked well with MGS1... Thanks for any help.

Version 1.5.0 removed native controller support from MGS2 and MGS3. The games now rely entirely on Steam Input, which is why your PS4 controller is no longer recognized by the game, since DS4Win just uses an Xinput wrapper to make games think you're using an Xbox controller.

I don't think you'll be getting any support from the devs for your pirated version, and I'm not really interested in helping you either, but I suppose I am curious, so this is all I'll say: you could try adding your version of the game to Steam as a "Non-Steam Game" using the button at the bottom-left of the library. As long as Steam Input is enabled, it should hook into the exe and _maybe_ controllers will work then.

**Comment by MultiKoopa** [comment]
Was uh... was this ever fixed?

**Comment by LethalPlacebo** [comment]
> Was uh... was this ever fixed?

There's nothing to fix. Controller support is now entirely handled via Steam Input. If you're running an illegitimate copy of the game, you're not running it through Steam, which means no Steam Input, which means no controller support.

**Comment by MultiKoopa** [comment]
> > Was uh... was this ever fixed?
> 
> There's nothing to fix. Controller support is now entirely handled via Steam Input. If you're running an illegitimate copy of the game, you're not running it through Steam, which means no Steam Input, which means no controller support.

Meaning if you disable Steam Input, it won't work, yeah? But if it's on, then it should work on any OS?

**Comment by LethalPlacebo** [comment]
> > > Was uh... was this ever fixed?
> > 
> > 
> > There's nothing to fix. Controller support is now entirely handled via Steam Input. If you're running an illegitimate copy of the game, you're not running it through Steam, which means no Steam Input, which means no controller support.
> 
> Meaning if you disable Steam Input, it won't work, yeah? But if it's on, then it should work on any OS?

That's right. Yeah it should work on any OS, but I assume you're referring to Linux/Steam Deck which I have no real experience with, so hopefully someone else here can give you some advice.

**Comment by MultiKoopa** [comment]
> > > > Was uh... was this ever fixed?
> > > 
> > > 
> > > There's nothing to fix. Controller support is now entirely handled via Steam Input. If you're running an illegitimate copy of the game, you're not running it through Steam, which means no Steam Input, which means no controller support.
> > 
> > 
> > Meaning if you disable Steam Input, it won't work, yeah? But if it's on, then it should work on any OS?
> 
> That's right. Yeah it should work on any OS, but I assume you're referring to Linux/Steam Deck which I have no real experience with, so hopefully someone else here can give you some advice.

took me a while to figure it out, since I've never seen it before, but apparently, for Steam Input on the Steam Deck for MGS, you have to map each button to "Game Actions", which include A/B/X/Y etc. very strange.

But it works now!

**Comment by Suzzbak06** [comment]
> > > > > Was uh... was this ever fixed?
> > > > 
> > > > 
> > > > There's nothing to fix. Controller support is now entirely handled via Steam Input. If you're running an illegitimate copy of the game, you're not running it through Steam, which means no Steam Input, which means no controller support.
> > > 
> > > 
> > > Meaning if you disable Steam Input, it won't work, yeah? But if it's on, then it should work on any OS?
> > 
> > 
> > That's right. Yeah it should work on any OS, but I assume you're referring to Linux/Steam Deck which I have no real experience with, so hopefully someone else here can give you some advice.
> 
> took me a while to figure it out, since I've never seen it before, but apparently, for Steam Input on the Steam Deck for MGS, you have to map each button to "Game Actions", which include A/B/X/Y etc. very strange.
> 
> But it works now!

Do you mind elaborating how you mapped each button to “Game Actions”? Where’s the menu for that?

**Comment by LethalPlacebo** [comment]
> > > > > > Was uh... was this ever fixed?
> > > > > 
> > > > > 
> > > > > There's nothing to fix. Controller support is now entirely handled via Steam Input. If you're running an illegitimate copy of the game, you're not running it through Steam, which means no Steam Input, which means no controller support.
> > > > 
> > > > 
> > > > Meaning if you disable Steam Input, it won't work, yeah? But if it's on, then it should work on any OS?
> > > 
> > > 
> > > That's right. Yeah it should work on any OS, but I assume you're referring to Linux/Steam Deck which I have no real experience with, so hopefully someone else here can give you some advice.
> > 
> > 
> > took me a while to figure it out, since I've never seen it before, but apparently, for Steam Input on the Steam Deck for MGS, you have to map each button to "Game Actions", which include A/B/X/Y etc. very strange.
> > But it works now!
> 
> Do you mind elaborating how you mapped each button to “Game Actions”? Where’s the menu for that?

Right-click the game in your Library > Manage > Controller Layout > Edit Layout

You should now see a screen showing your controller buttons on the left and what they do on the right. Click the rectangular box on the right and you'll be taken to a screen that lets you choose the buttons. At the top, you'll have various tabs that allow you to choose what type of button is mapped. To the far left is "Game Actions" and it appears that the MGS games must use this tab to map functions or they just don't work.
